src/Symfony/Component/Form/Tests/Extension/Core/Type/TextTypeTest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Tests\Extension\Core\Type;
+
+use Symfony\Component\Form\Test\TypeTestCase as TestCase;
+
+class TextTypeTest extends TestCase
+{
+    public function testSubmitNullReturnsNull()
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\TextType', 'name');
+
+        $form->submit(null);
+
+        $this->assertNull($form->getData());
+    }
+
+    public function testSubmitNullReturnsEmptyStringWithEmptyDataAsString()
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\TextType', 'name', array(
+            'empty_data' => '',
+        ));
+
+        $form->submit(null);
```

Comment:
@backbone87 I don't really understand what is "new" for you, as I am feeling quite new myself in symfony :)

1. But as far as I understand when you call `Form::submit(null)` here what happens:
  - [view data is initialized with null](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L535),
  - if not `compound` and not `inherit_data` (e.g `TextType`) [view data is set to submitted data](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L595) which [may be `null` and would not be converted to empty string](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L528)

2. So at this point view data cannot be an empty string, it remains `null`.

  Then:
  - [view data is assigned to `empty_data`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L605), either simply set or a [callable getting the form and  `null`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L602)
  - [the final resolution](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L600) as before anything we must be sure that [if compound data must be an array, if not it must be a string](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Extension/Core/Type/FormType.php#L141).

3. Hence here we do have a view data as empty string but...:

  - [here comes the devil](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L622) because before getting to the model data [the normalization set it to null](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L1123) :/,
  - `Form::viewToNorm()` is actually the default view data transformer,

Conclusion
========
This PR could add a view data transformer to [prevent that call](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L1122) but I guess in order to prevent messing with other transformers and as this has to stay the default, @webmozart asked to add it as a model data transformer since it is the first registered => also the last called => minimum of potential BC breaks.